servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -109,13 +117,14 @@ public final class RetryingHttpRequesterFilter
 
     RetryingHttpRequesterFilter(
             final boolean waitForLb, final boolean ignoreSdErrors, final boolean mayReplayRequestPayload,
-            final int maxTotalRetries,
+            final boolean returnFailedResponses, final int maxTotalRetries,
             @Nullable final Function<HttpResponseMetaData, HttpResponseException> responseMapper,
             final BiFunction<HttpRequestMetaData, Throwable, BackOffPolicy> retryFor,
```

Comment:
It seems like we may leak "pending response" if any user-defined function under `retryFor` throws. Scenario:
1. Server responds with 503
2. User maps it to `HttpResponseException`
3. returnFailedResponses == true, we don't drain proactively
4. `OuterRetryStrategy.apply` is invoked
5. `retryFor` throws
6. We won't invoke `applyRetryCallbacks` and won't drain that response.

Consider adding a try-catch inside `apply` to make sure we train `t instance of HttpResponseException` in case of any unexpected exception.

Adding a test will be highly appreciated.